### PR TITLE
osd_activate: fix osd umount on container stop

### PR DESCRIPTION
**Pull request by guits**
There's no need to have complexity to get the path of the current OSD.

By the way, at the moment this complexity leads to incorrect behavior:

```
-bash-4.2# docker exec ceph-osd-2 findmnt --nofsroot --noheadings --output SOURCE --submounts --target /var/lib/ceph/osd/
/dev/mapper/atomicos-root
/dev/sdb1
/dev/sda1
-bash-4.2# docker exec ceph-osd-2 df --output=target | grep '/var/lib/ceph/osd/'
/var/lib/ceph/osd/ceph-0
/var/lib/ceph/osd/ceph-2
-bash-4.2#
```

In the first method used in recent branches, it means we might try to umount
the root partition.
In the second method used in older branches we try to umount something
messed up:
```
2021-01-14 00:17:08  /opt/ceph-container/bin/entrypoint.sh: osd_disk_activate: Unmounting /var/lib/ceph/osd/ceph-0
/var/lib/ceph/osd/ceph-2

...

umount: /var/lib/ceph/osd/ceph-0
/var/lib/ceph/osd/ceph-2: mountpoint not found
++/opt/ceph-container/bin/osd_disk_activate.sh:92: sigterm_cleanup_post(): log 'osd_disk_activate: Failed to umount /var/lib/ceph/osd/ceph-0
/var/lib/ceph/osd/ceph-2'

...

++/opt/ceph-container/bin/common_functions.sh:13: log(): echo '2021-01-14 00:17:08  /opt/ceph-container/bin/entrypoint.sh: osd_disk_activate: Failed to umount /var/lib/ceph/osd/ceph-0
/var/lib/ceph/osd/ceph-2'

```
We can easily build and predict the path to umount since we have
`$CLUSTER` and `$OSD_ID` variables in the code of `osd_disk_activate.sh`

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1921750

Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

**Comment by guits** [comment]
jenkins test ceph_ansible-lvm_osds

**Comment by guits** [comment]
jenkins test ceph_ansible-lvm_osds

**Comment by guits** [comment]
jenkins test ceph_ansible-all_daemons

**Comment by guits** [comment]
jenkins test ceph_ansible-collocation

**Comment by guits** [comment]
jenkins test ceph_ansible-lvm_osds

File: src/daemon/osd_scenarios/osd_disk_activate.sh [hunk 1]
```diff
@@ -22,35 +22,37 @@ function osd_activate {
...
-  if ! ceph-volume simple activate --file ${CEPH_VOLUME_SCAN_FILE} --no-systemd; then
+  if ! ceph-volume simple activate --file "${CEPH_VOLUME_SCAN_FILE}" --no-systemd; then
     cat /var/log/ceph
     exit 1
   fi
 
+  if [[ ${OSD_DMCRYPT} -eq 1 ]]; then
+    umount_lockbox
+  fi
+
```
Comment (dsavineau) [inline src/daemon/osd_scenarios/osd_disk_activate.sh:45]: shouldn't we do that before the ceph-volume simple activate ?

File: src/daemon/osd_scenarios/osd_volume_activate.sh [hunk 2]
```diff
@@ -95,18 +95,20 @@ function osd_volume_activate {
     osd_volume_simple
   fi
 
+  if [[ ${OSD_DMCRYPT} -eq 1 ]]; then
+    umount_lockbox
+  fi
+
```
Comment (dsavineau) [inline src/daemon/osd_scenarios/osd_volume_activate.sh:101]: this should be done only if OSD_VOLUME_TYPE==simple so probably need to integrate this statement after the osd_volume_simple call

**Review by dsavineau — CHANGES_REQUESTED** [review]

**Comment by guits** [comment]
jenkins test ceph_ansible-lvm_osds

**Comment by guits** [comment]
jenkins test ceph_ansible-lvm_osds

**Comment by guits** [comment]
jenkins test ceph_ansible-lvm_osds

**Comment by dsavineau** [comment]
jenkins test ceph_ansible-lvm_osds

**Review by dsavineau — CHANGES_REQUESTED** [review]

File: src/daemon/osd_scenarios/osd_volume_activate.sh [hunk 3]
```diff
@@ -93,6 +94,9 @@ function osd_volume_activate {
     osd_volume_lvm
   else
     osd_volume_simple
+    if [[ ${OSD_DMCRYPT} -eq 1 ]]; then
+      umount_lockbox
+    fi
```
Comment (guits) [inline src/daemon/osd_scenarios/osd_volume_activate.sh:99]:
```suggestion
```

Comment (dsavineau) [reply src/daemon/osd_scenarios/osd_volume_activate.sh:99]: :+1:

**Comment by guits** [comment]
jenkins test ceph_ansible-lvm_osds

**Review by dsavineau — APPROVED** [review]

**Pull request merged**